File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.cpp
class Solution {
   public:
    int evalRPN(vector<string>& tokens) {
        int result = stoi(tokens[0]);
        int const tokensSize = static_cast<int>(tokens.size());
        stack<int> s;

        if (tokensSize == 1) {
            return result;
        }

        for (int i = 0; i < tokensSize; ++i) {
            if (tokens[i] == "+") {
                int const a = s.top();
                s.pop();
                int const b = s.top();
                s.pop();
                result = (b + a);
                s.push(result);
            } else if (tokens[i] == "-") {
                int const a = s.top();
                s.pop();
                int const b = s.top();
                s.pop();
                result = (b - a);
                s.push(result);
            } else if (tokens[i] == "*") {
                int const a = s.top();
                s.pop();
                int const b = s.top();
                s.pop();
                result = (b * a);
                s.push(result);
            } else if (tokens[i] == "/") {
                int const a = s.top();
                s.pop();
                int const b = s.top();
                s.pop();
                result = (b / a);
                s.push(result);
            } else {
                s.push(stoi(tokens[i]));
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/reconstruct-flight-path/submission-1.py
class Solution:
    def findItinerary(self, tickets: List[List[str]]) -> List[str]:
        graph = defaultdict(list)
        result = []

        for start, end in tickets:
            heapq.heappush(graph[start], end)

        stack = ["JFK"]

        while stack:
            while graph[stack[-1]]:
                stack.append(heapq.heappop(graph[stack[-1]]))
            result.append(stack.pop())

        result.reverse()
        return result
